### PR TITLE
Feature/react on rails

### PR DESCRIPTION
**Pull request by micapam**
Based on advice from Dan Draper I've switched out `react-rails` gem for the `react-on-rails` gem.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -8,8 +8,9 @@ source 'https://rubygems.org'
 # Core gems
 gem 'rails', '>= 5.0.0.beta4', '< 5.1'
 gem 'pg', '>= 0.19.0.beta'
-gem 'react-rails', '~> 1.5'
-gem 'jquery-rails'
```
Comment (alecky) [inline Gemfile]: good to see jquery out!!

File: client/app/bundles/HelloWorld/containers/HelloWorld.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  const { updateName } = actions;
+  const name = $$helloWorldStore.get('name');
+
+  // This uses the ES2015 spread operator to pass properties as it is more DRY
+  // This is equivalent to:
+  // <HelloWorldWidget $$helloWorldStore={$$helloWorldStore} actions={actions} />
+  return (
+    <HelloWorldWidget {...{ updateName, name }} />
+  );
+};
```
Comment (alecky) [inline client/app/bundles/HelloWorld/containers/HelloWorld.jsx]: ive seen more elegant ways to create smart components where you can map state to props and dispatcher to props.

File: client/package.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  },
+  "scripts": {
+    "build:test": "webpack --config webpack.config.js",
+    "build:production": "NODE_ENV=production webpack --config webpack.config.js",
+    "build:development": "webpack -w --config webpack.config.js"
+  },
+  "cacheDirectories": ["node_modules", "client/node_modules"],
+  "dependencies": {
+    "babel": "^6.5.2",
+    "babel-cli": "^6.6.5",
```
Comment (alecky) [inline client/package.json]: is the cli necessary

File: client/package.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    "imports-loader": "^0.6.5",
+    "mirror-creator": "1.1.0",
+    "react": "^0.14.8 || ^15.0.0",
+    "react-dom": "^0.14.8 || ^15.0.0",
+    "react-on-rails": "6.0.5",
+    "react-redux": "^4.4.1",
+    "redux": "^3.3.1",
+    "redux-promise": "^0.5.3",
+    "redux-thunk": "^2.0.1",
+    "webpack": "^1.12.14"
```
Comment (alecky) [inline client/package.json]: shouldn't this be a dev dependancy

File: client/app/bundles/HelloWorld/containers/HelloWorld.jsx (hunk 2 above)
Comment (micapam) [inline client/app/bundles/HelloWorld/containers/HelloWorld.jsx]: Would be glad to hear them! Though this is dummy example code auto-generated by installer, so no point changing them in this actual case.

**Pull request merged**